sixtyfps_runtime/corelib/api.rs

```diff
@@ -0,0 +1,276 @@
+    pub fn show(&self) {
+        self.0.show();
+    }
+
+    /// De-registers the window from the windowing system, therefore hiding it.
+    pub fn hide(&self) {
+        self.0.hide();
+    }
+}
+
+impl crate::window::WindowHandleAccess for Window {
+    fn window_handle(&self) -> &Rc<crate::window::Window> {
+        &self.0
+    }
+}
+/// This trait is used to obtain references to global singletons exported in `.60`
+/// markup. Alternatively, you can use [`ComponentHandle::global`] to obtain access.
+///
+/// This trait is implemented by the compiler for each global singleton that's exported.
+///
+/// # Example
+/// The following example of `.60` markup defines a global singleton called `Palette`, exports
+/// it and modifies it from Rust code:
+/// ```rust
+/// # sixtyfps_rendering_backend_testing::init();
+/// sixtyfps::sixtyfps!{
+/// export global Palette := {
+///     property<color> foreground-color;
+///     property<color> background-color;
+/// }
+///
+/// export App := Window {
+///    background: Palette.background-color;
+///    Text {
+///       text: "Hello";
+///       color: Palette.foreground-color;
+///    }
+///    // ...
+/// }
+/// }
+/// let app = App::new();
+/// app.global::<Palette>().set_background_color(sixtyfps::Color::from_rgb_u8(0, 0, 0));
+///
+/// // alternate way to access the global singleton:
+/// Palette::get(&app).set_foreground_color(sixtyfps::Color::from_rgb_u8(255, 255, 255));
+/// ```
+///
+/// See also the [language reference for global singletons](docs/langref/index.html#global-singletons) for more information.
+pub trait Global<'a, Component> {
+    /// Returns a reference that's tied to the life time of the provided component.
+    fn get(component: &'a Component) -> Self;
+}
+
+/// This trait describes the common public API of a strongly referenced SixtyFPS component.
+/// It allows creating strongly-referenced clones, a conversion into/ a weak pointer as well
+/// as other convenience functions.
+///
+/// This trait is implemented by the [generated component](mod@crate#generated-components)
+pub trait ComponentHandle {
+    /// The type of the generated component.
+    #[doc(hidden)]
+    type Inner;
+    /// Returns a new weak pointer.
+    fn as_weak(&self) -> Weak<Self>
+    where
+        Self: Sized;
+
+    /// Returns a clone of this handle that's a strong reference.
+    #[must_use]
+    fn clone_strong(&self) -> Self;
+
+    /// Internal function used when upgrading a weak reference to a strong one.
+    #[doc(hidden)]
+    fn from_inner(_: vtable::VRc<ComponentVTable, Self::Inner>) -> Self;
+
+    /// Marks the window of this component to be shown on the screen. This registers
+    /// the window with the windowing system. In order to react to events from the windowing system,
+    /// such as draw requests or mouse/touch input, it is still necessary to spin the event loop,
+    /// using [`crate::run_event_loop`].
```

Comment:
I wonder if this will still work properly.
Will it work if you do `[sixtyfps::run_event_loop`].  I don't think so.

Reply:
Indeed, this link and the others are broken. /me digs

Reply:
I have not been able to solve this without either

   (1) Having the real implementation of `run_event_loop` in corelib, but then it would `panic!` if no backend was initialised yet.

    (2) using a HTML link (`fn.run_event_loop.html`)

Hacks using `#[cfg(doc)]` with conditional exporting in `corelib` didn't work because they were not in place anymore when generating docs for the API crate.

So I've opted for (2) for now.